### PR TITLE
Initialize(), Implicit var/, Addtimer() and Hacky Code update to Contributing.md

### PR DESCRIPTION
**Pull request by variableundefined**
## What Does This PR Do
This adds instructions on our preference on initialize() over new() (From TG), no implicit var in proc (original), addtimer() (Also original but seems to be not in tg's contributing.md for some reason), and one more sentence to the section on hacky code to explain how to bugfix properly instead of bandaiding.

## Why It's Good For The Game

Code quality

## Changelog
No CL

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -236,6 +238,24 @@ There are two key points here:
 
 Remember: although this tradeoff makes sense in many cases, it doesn't cover them all. Think carefully about your addition before deciding if you need to use it.
 
+### Prefer `Initialize()` over `New()` for atoms
+Our game controller is pretty good at handling long operations and lag, but it can't control what happens when the map is loaded, which calls `New` for all atoms on the map. If you're creating a new atom, use the `Initialize` proc to do what you would normally do in `New`. This cuts down on the number of proc calls needed when the world is loaded. See here for details on `Initialize`: https://github.com/tgstation/tgstation/blob/master/code/game/atoms.dm#L49
```
Comment (Couls) [inline .github/CONTRIBUTING.md]: Probably want to change the link to not link to TG

Comment (variableundefined) [reply .github/CONTRIBUTING.md]: Done. I just removed the link since the link was outdated itself

Comment (Couls) [reply .github/CONTRIBUTING.md]: here's the link to our comments on the initialize proc if you want to replace it https://github.com/ParadiseSS13/Paradise/blob/f1de76e50020975e34f1d6039a1047b7e3a24ec8/code/game/atoms.dm#L71

Comment (variableundefined) [reply .github/CONTRIBUTING.md]: I think it's better for them to use Ctrl + T with VSC and then do so

**Pull request merged**